py3-cython.yaml

```diff
@@ -40,17 +40,39 @@ subpackages:
       provider-priority: ${{range.value}}
       provides:
         - ${{package.name}}
-        - cython
     pipeline:
       - uses: py/pip-build-install
         with:
           python: python${{range.key}}
       - uses: strip
 
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}-bin
+    description: Executable binaries for ${{vars.pypi-package}} installed for python${{range.key}}
+    dependencies:
+      runtime:
+        - py${{range.key}}-${{vars.pypi-package}}-bin
+      provides:
+        - py3-${{vars.pypi-package}}
+        - cython
```

Comment:
Tagged this on the other PR but this shouldn't provide the models just the binaries.

Reply:
`py3-build.yaml` has the provides here to be `py3-${{vars.pypi-package}}` for the bin subpackage. I don't know if we might want this to be

`cython=${{package.full-version}}" as we have a `cython-0` version stream also

Reply:
I think we want `cython-bin` subpackages to be dependent on the corresponding non-bin subpackages.   And we want the `-bin` packages to have the provides for `cython` that way installing the modules doesn't necessarily install the binaries but installing a specific binary version gets you the corresponding modules.

I don't think we want to provide a new stream though.  No-one should be targeting a specific stream other than the 0 stream and it's not being used by anything anymore. The package it was added for doesn't have problems building with newer versions of cython.

Reply:
How would a multi-versioned py3-foo.yaml use cython?

As I understand it, something using cython is going to need a py3.XX-cython and a /usr/bin/cython.
I ran into this just recently in py3-maturin also.

How about this:
1. make py3-cython package build a non-versioned '/usr/bin/cython' with the package name 'cython'
2. remove the cython-0 package
3. make py3.XX-cython depend on 'cython'

Then, to use cython in a build the package just needs to add `py3-cython` or `py3-supported-cython`.

Reply:
> How would a multi-versioned py3-foo.yaml use cython?

The same way they do now by having a dep on cython? This only changes if their package imports cython from the building interpreter not if they use a compiler. 

> remove the cython-0 package

I missed 2 packages that are using cython~0 so getting rid of it might not be possible right now. I'll check if they can still need that version.

> Then, to use cython in a build the package just needs to add py3-cython or py3-supported-cython

We already have people depending more people depending on `cython` than `py3-cython` and I don't see an advantage to having a real `cython` package rather than bin packages that provide `cython` and letting apk sort out which one to use. As I understand it the dependency solver will choose based on package priority so it's already going to end up with the newest interpreter. I do see potential advantage to being able to create an environment with only a single python interpreter and all the expected pieces available.